### PR TITLE
Fix single result page test flakiness

### LINKED ISSUE #3112: Flaky test: visual-regression/pages/pages-single-result.spec.ts:27:11 › screen at breakpoint lg with width 1024 › image rtl single-result page snapshots from search results
## Description

This test failed on main: https://github.com/WordPress/openverse/actions/runs/6382150807/job/17321100648

The media details should be in a horizontal row for breakpoints above sm. However, when creating the Playwright snapshots, the layout changes to vertical (as in the xs breakpoint):

image-rtl-from-search-results-lg-actual

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3112 by @obulat

The changed files are mostly the snapshots: I added the single result page snapshots with the `additional_search_flag` on because we will start implementing the changes soon.

## Description

The `use-layout` composable watches the page width and updates the `breakpoint` property in the `ui` store if the page width changes.
We used `throttle` to prevent frequent updates from affecting the site performance. However, throttled watch meant that we might not update the breakpoint to the last screen width. 

I think that when taking the screenshot, the screen width is lowered, the breakpoint is updated so the layout is changed to `xs` layout (so, the metadata is shown as a vertically), and then the breakpoint is not updated to the value that the screen ends up with (the non-xs) - so the metadata stays vertical instead of being horizontal for `sm` and up.

~~This PR also adds the single result page visual regression tests for when `additional_search_views` flag is on. In this PR, they are the same as the `off` snapshots, but they will be updated in #2933 for audio and in #3164 for images.~~ This change was removed to make a `critical` PR easier to review following @krysal's [suggestion](https://github.com/WordPress/openverse/pull/3186#pullrequestreview-1674660558).

This PR also adds several changes to the tests to speed them up (because one of the CI runs failed because the tests were taking too long): blocks the audio thumb requests to speed up the loading, adds thumb tapes to prevent network requests, stops covering the images because we are using the saved tapes and the images will be loaded well in the CI also.

The CI [failure](https://github.com/WordPress/openverse/actions/runs/6485793925?pr=3186) was caused by the thumbnail being slightly different from the actual image. Previously, the image from the source was used. In this PR, it uses the thumbnail saved in a tape.

Another CI failure was due to the navigation and URL change not being fully synced. I was using `waitForUrl` expecting that after this call, the home page would be replaced by the search page. However, in the traces, you can see that the URL was updated even though homepage was still visible. This is why the tests for the search page were failing.

After the latest changes, I ran the CI 7 times, and it failed once due to a timeout (we only allow 15 minutes for the VR tests, and it took longer and was canceled this time). I suspect that this timeout issue happens when we have an API response time spike. It would be nice to investigate what requests are being made during a Playwright VR test run.

## Testing Instructions

I will re-run the CI several times to check that there is no flakiness.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by obulat** [comment]
> **Playwright failure test results**: https://github.com/WordPress/openverse/actions/runs/6482175402
> 
> It looks like some of the Playwright tests failed. You can download the trace output and image diffs for the failed tests under the "Artifacts" section in the above page.
> 
> You can read the [docs on how to use this artifact](https://docs.openverse.org/frontend/reference/playwright_tests.html#debugging).
> 
> If the test is flaky, follow the [flaky test triage procedure](https://docs.openverse.org/general/test.html#flaky-tests).

This failure was caused by long running tests and not by the failure of snapshot tests.

File: frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts [hunk 1]
```diff
@@ -7,44 +7,52 @@ import {
...
+          await setCookies(page.context(), {
+            features: { additional_search_views: isOn ? "on" : "off" },
+          })
+          await preparePageForTests(page, breakpoint)
+          await page.route("**", (route) => {
+            const url = route.request().url()
+            // For audio, use the generated image instead of requesting the
+            if (
+              url.endsWith(".jpg") ||
+              (url.endsWith("/thumb/") && url.includes("/audio/"))
```
Comment (dhruvkb) [inline frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts:33]: Nit: regex match here could prevent three separate conditions.

**Review by dhruvkb — APPROVED** [review]
Thanks for finding the cause and fixing the tests. It was indeed a very strange and hard to reproduce flake.

**Review by krysal — COMMENTED** [review]
Very nice finding the problem and a fix! I'm not a requested reviewer, but I see a lot going on here, so thank you for describing each issue!

> This PR also adds the single result page visual regression tests for when additional_search_views flag is on.

Since this is labeled as `critical` is this addition necessary as part of the fix? Can it be left for another pull request? Especially since it will be modified in #2933  and #3164. Reviewing it will be easier, narrowing the solution to the problem.

Regarding the last paragraph:

> After the latest changes, I ran the CI 7 times, and it failed once due to a timeout (we only allow 15 minutes for the VR tests, and it took longer and was canceled this time). I suspect that this timeout issue happens when we have an API response time spike. It would be nice to investigate what requests are being made during a Playwright VR test run.

Do you think it warrants to create an investigation issue?

File: frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts [hunk 2]
```diff
@@ -7,44 +7,52 @@ import {
...
-          { maxDiffPixelRatio: 0.01 }
-        )
+        }) => {
+          await setCookies(page.context(), {
+            features: { additional_search_views: isOn ? "on" : "off" },
+          })
+          await preparePageForTests(page, breakpoint)
+          await page.route("**", (route) => {
+            const url = route.request().url()
+            // For audio, use the generated image instead of requesting the
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts]: I'm not sure if it's worth taking more space by saving more tapes for audio covers to make the snapshots closer to the real flow. For images, the image grid changes based on the real dimensions of the images, but for audio, the thumbnails don't change the layout, so the changes would be visual only.

[the remark this answers is not in the document]
Comment (krysal) [reply frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts]: Makes sense. Thanks!

**Review by krysal — APPROVED** [review]
Great!

**Pull request merged**